apps/server/src/modules/user/legacy/controller/api-test/admin-api-teachers.api.spec.ts

```diff
@@ -0,0 +1,282 @@
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(teacherUser1, teacherAccount1);
+			};
+
+			it('should reject request', async () => {
+				setup();
+				await request(app.getHttpServer()) //
+					.get(`${basePath}/${teacherUser1.id}`)
+					.expect(403);
+			});
+		});
+
+		describe('when teacher does not exists', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+			};
+
+			it('should reject request ', async () => {
+				setup();
+				await request(app.getHttpServer()) //
+					.get(`${basePath}/000000000000000000000000`)
+					.expect(404);
+			});
+		});
+	});
+
+	describe('[GET]', () => {
+		describe('when sort param is provided', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 0,
+					$limit: 5,
+					$sort: { firstName: 1 },
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return teachers in correct order', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.set('Accept', 'application/json')
+					.expect(200);
+
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(2);
+				expect(data.length).toBe(2);
+				expect(data[0]._id).toBe(teacherUser1._id.toString());
+				expect(data[1]._id).toBe(teacherUser2._id.toString());
+			});
+		});
+
+		describe('when search params are too tight', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 0,
+					$limit: 5,
+					$sort: { firstName: 1 },
+					classes: ['1A'],
+					consentStatus: { $in: ['ok'] },
+					createdAt: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+					lastLoginSystemChange: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+					outdatedSince: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return empty list', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.send()
+					.expect(200);
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(0);
+				expect(data.length).toBe(0);
+			});
+		});
+
+		describe('when skip params are too big', () => {
```

Comment:
This test only exists for teachers and not for students, but they should behave the same, shouldn't they?

It's also a test where I'm not sure if it has to be an API test or if it is enough to test it on repo level. I tend to say the latter. Same for skipping and sorting.

If you want to keep it as an API test, the expectation should be more specific like "should return an empty list".

Reply:
This test is so general as test database seems to omit skip and limit params. It work perfectly fine on real databse

Reply:
If this is the case we should look into it. "seems to omit" is not a good reason for a decision.

Reply:
My bad, I was validating total number which return numbers of all users without skip and limit conditions